java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -378,7 +379,7 @@ protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException
 
         for (int i = 0; i < count; i++) {
           parser.nextToken();
-          BigInteger value = parser.getBigIntegerValue();
+          BigInteger value = new BigInteger(parser.getValueAsString());
```

Comment:
will this not overflow doing a biginteger.longValue()? or change sign.

Reply:
I guess not, when write UINT8 values with JsonFileReader, it uses getNoOverflow which is already BigInteger?

Reply:
no i mean the line below "buf.writeLong(value.longValue())"

Reply:
Seems not, I will take a close watch.

Reply:
Ok went through UINT8Vector and saw the method implementation, i think there should be no problems like you said too.(although a test case would be nice, nothing blocking though).

Reply:
Sure, I could add some tests for UINT type vectors in a seperate issue :)